### DISCUSSION TITLE
Enhanced Note Location

### DISCUSSION THREAD

**Discussion by uroybd** [oleeskild/obsidian-digital-garden#157] [Ideas]
Currently, the note URL is being determined by the `dg-permalink` key, and in its absence, by location in the vault. A thing to notice here is, no matter what the permalink is, it will always be placed in the file tree as in the vault.

Another thing is, a gardener might want a different organization in the published garden which can differ from the vault.

To achieve more control over the representations I thought of two solutions which will complement each other.

### Path Rewrite Rules
A textarea in the settings where the users can add rewrite rules. For example:

```
Personal/Journal:Personal
Personal/Books/Read:What I Read
```
In every line of colon-delimited rules, the first part is to be the `from` part, and the second part is to be the `to` part.

### dg-path
A frontmatter entry to provide the path of the note.

### Precedences in Path resolution
1. dg-path
2. default path with rewrite rules applied

### Precedences for URL generation
1. dg-permalink
2. permalink based path as resolved by path resolution.

**Comment by Mara-Li** [comment]
I would love to see this feature!

**Comment by oleeskild** [comment]
Great suggestion! I've found myself wanting to have a different structure in my vault than in the published site. I think the implementation suggested here would be great, and allow for lots of customisations without getting overly complicated.

**Reply by uroybd** [reply]
I'll try to implement it then!

**Reply by Mara-Li** [reply]
Hello! Any news about it?

**Reply by uroybd** [reply]
I will work on it sometimes soon.

**Reply by iwyatt** [reply]
+1 

Currently I have an entirely separate vault for my digital garden simply because the path mirrors the vault structure, which differs from how I would want to organize it if I kept everything within my primary vault.

**Reply by uroybd** [reply]
I've implemented it. It's awaiting approval.

**Reply by oleeskild** [reply]
Great, thank you @uroybd. I'll try to get to it today

**Reply by uroybd** [reply]
@oleeskild Wait a bit, I'm doing a little more enhancement.

**Reply by uroybd** [reply]
> @oleeskild Wait a bit, I'm doing a little more enhancement.

I tried, but it seems impossible, you can merge the current version.
